### PR TITLE
func: port to riscv

### PR DESCRIPTION
**Pull request by Kepontry**
Port this to riscv, use the "time" CSR reg as the counter, tested on the JH7110 SoC(riscv64, 4-core SiFive U74 inside) of VisionFive2 SBC.

### CODE REVIEW

**Comment by fabiensanglard** [comment]
Thanks for this contribution. However putting all that in `clock.c` hurts readability. The best way is to plage this code in `clock_risc.c` and leverage the linker in `Makefile`.

**Comment by fabiensanglard** [comment]
This is hard to follow because the second commit undo what the first one did. Pleas squash them.

**Comment by Kepontry** [comment]
Sorry for the inconvenience. How about this version?
